### PR TITLE
data_schema, ports interface definition, ...

### PR DESCRIPTION
**Pull request by nicolasnk**
Implemented schema definition of our data in protobuffer and started to define interface in ports and their adapeters. Could not generate go code from the proto files because I need to fix an issue with the protoc compiler that I have installed on my computer. Will be done in the upcoming days.

### CODE REVIEW

File: useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+message DateOfBirth {
+  string day = 1;
+  string month = 2;
+  string year = 3;
+}
+
+// might some performance consequence but I will into it later
+message UIUD {
```
Comment (elmas23) [inline useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto]: nit: typo UUID

File: useraccount/internal/ports/driverframework.go [hunk 2]
```diff
@@ -1 +1,27 @@
...
+}
+
+ type FindPersonKey struct {
+
+}
+
+type APIPort interface {
+	AddNewPersonInfo(person Person) error
+	EditPersonInfo(pid PersonId) error
+	DeletePersonInfo(pid PersonId) error
```
Comment (elmas23) [inline useraccount/internal/ports/driverframework.go]: We agreed on not having Delete

File: useraccount/internal/ports/drivenframework.go [hunk 3]
```diff
@@ -1 +1,10 @@
 package ports
+
+type PostgresSQLPort interface {
+	OpenDBConnection()
+	CloseDBConnection()
+	AddNewPersonInfo()
+	DeletePersonInfo()
```
Comment (elmas23) [inline useraccount/internal/ports/drivenframework.go:7]: Not going to implement

File: useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  string uiud = 1;
+}
+
+message PersonInfo {
+  Names names = 1;
+  Biometric biometrics = 2;
+  Address address = 3;
+  Origin origins = 4;
+  Phenotype phenotypes = 5;
+  DateOfBirth date_of_birth = 6;
```
Comment (elmas23) [inline useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto:59]: nit: can we stick to camelCase for everything: dateOfBirth ?

File: useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+message Origin {
+  repeated string province = 1; // could be from multiple provinces
+  string chef_lieu = 2;
+}
+
+message Phenotype {
+  float height = 1;
+  float weight = 2;
+  string eye_color = 3;
```
Comment (elmas23) [inline useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto:32]: same: camelCase eyeColor

File: useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+message Names {
+   string nom = 1;
+   string post_nom = 2;
+   repeated string middle_names = 3;
+}
+
+message Origin {
+  repeated string province = 1; // could be from multiple provinces
+  string chef_lieu = 2;
```
Comment (elmas23) [inline useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto:26]: chefLieu

File: useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+package pb;
+
+option go_package = "./pb";
+
+message Address {
+  int32 number = 1;
+  string avenue = 2;
+  string quartier = 3;
+  string commune = 4; // we can further make this of type Enum since we know all the possible communes
+  string zip_code = 5;
```
Comment (elmas23) [inline useraccount/internal/adapters/framework/driver/grpc/proto/personInfo_msg.proto:12]: camelCase

**Review by elmas23 — CHANGES_REQUESTED** [review]
Can we stick to camelCase naming convention for file, variable ?

all files should be camelcase except for test file since we add _test at the end.

File: useraccount/internal/ports/driverframework.go (hunk 2 above)
Comment (nicolasnk) [reply useraccount/internal/ports/driverframework.go]: oh yeah it makes sense 
gonna remove it after

File: useraccount/internal/ports/drivenframework.go (hunk 3 above)
Comment (nicolasnk) [reply useraccount/internal/ports/drivenframework.go:7]: Not sure I understand the comment

Comment (elmas23) [reply useraccount/internal/ports/drivenframework.go:7]: We are not going to implement delete CRUD

**Review by elmas23 — APPROVED** [review]

**Pull request merged**